File: src/main/java/com/refinedmods/refinedstorage/item/WirelessFluidGridItem.java
package com.refinedmods.refinedstorage.item;

import com.refinedmods.refinedstorage.RS;
import com.refinedmods.refinedstorage.api.network.grid.IGrid;
import com.refinedmods.refinedstorage.api.network.item.INetworkItem;
import com.refinedmods.refinedstorage.api.network.item.INetworkItemManager;
import com.refinedmods.refinedstorage.apiimpl.network.item.WirelessFluidGridNetworkItem;
import com.refinedmods.refinedstorage.apiimpl.network.node.GridNetworkNode;
import com.refinedmods.refinedstorage.inventory.player.PlayerSlot;
import net.minecraft.world.entity.player.Player;
import net.minecraft.world.item.Item;
import net.minecraft.world.item.ItemStack;

import javax.annotation.Nonnull;

public class WirelessFluidGridItem extends NetworkItem {
    private final Type type;

    public WirelessFluidGridItem(Type type) {
        super(new Item.Properties().tab(RS.CREATIVE_MODE_TAB).stacksTo(1), type == Type.CREATIVE, () -> RS.SERVER_CONFIG.getWirelessFluidGrid().getCapacity());

        this.type = type;
    }

    public static int getSortingType(ItemStack stack) {
        return (stack.hasTag() && stack.getTag().contains(GridNetworkNode.NBT_SORTING_TYPE)) ? stack.getTag().getInt(GridNetworkNode.NBT_SORTING_TYPE) : IGrid.SORTING_TYPE_QUANTITY;
    }

    public static int getSortingDirection(ItemStack stack) {
        return (stack.hasTag() && stack.getTag().contains(GridNetworkNode.NBT_SORTING_DIRECTION)) ? stack.getTag().getInt(GridNetworkNode.NBT_SORTING_DIRECTION) : IGrid.SORTING_DIRECTION_DESCENDING;
    }

    public static int getSearchBoxMode(ItemStack stack) {
        return (stack.hasTag() && stack.getTag().contains(GridNetworkNode.NBT_SEARCH_BOX_MODE)) ? stack.getTag().getInt(GridNetworkNode.NBT_SEARCH_BOX_MODE) : IGrid.SEARCH_BOX_MODE_NORMAL;
    }

    public static int getTabSelected(ItemStack stack) {
        return (stack.hasTag() && stack.getTag().contains(GridNetworkNode.NBT_TAB_SELECTED)) ? stack.getTag().getInt(GridNetworkNode.NBT_TAB_SELECTED) : -1;
    }

    public static int getTabPage(ItemStack stack) {
        return (stack.hasTag() && stack.getTag().contains(GridNetworkNode.NBT_TAB_PAGE)) ? stack.getTag().getInt(GridNetworkNode.NBT_TAB_PAGE) : 0;
    }

    public static int getSize(ItemStack stack) {
        return (stack.hasTag() && stack.getTag().contains(GridNetworkNode.NBT_SIZE)) ? stack.getTag().getInt(GridNetworkNode.NBT_SIZE) : IGrid.SIZE_STRETCH;
    }

    public Type getType() {
        return type;
    }

    @Nonnull
    @Override
    public INetworkItem provide(INetworkItemManager handler, Player player, ItemStack stack, PlayerSlot slot) {
        return new WirelessFluidGridNetworkItem(handler, player, stack, slot);
    }

    public enum Type {
        NORMAL,
        CREATIVE
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/item/WirelessFluidGridItem.kt
package com.refinedmods.refinedstorage.item

import com.refinedmods.refinedstorage.RS
import com.refinedmods.refinedstorage.api.network.grid.IGrid
import com.refinedmods.refinedstorage.api.network.item.INetworkItem
import com.refinedmods.refinedstorage.api.network.item.INetworkItemManager
import com.refinedmods.refinedstorage.apiimpl.network.item.WirelessFluidGridNetworkItem
import com.refinedmods.refinedstorage.apiimpl.network.node.GridNetworkNode
import net.minecraft.entity.player.PlayerEntity
import net.minecraft.item.ItemStack
import java.util.function.Supplier

class WirelessFluidGridItem(val type: Type) : NetworkItem(Properties().group(RS.MAIN_GROUP).maxStackSize(1), type == Type.CREATIVE, Supplier { RS.SERVER_CONFIG.wirelessFluidGrid.getCapacity() }) {
    enum class Type {
        NORMAL, CREATIVE
    }

    @Nonnull
    override fun provide(handler: INetworkItemManager?, player: PlayerEntity?, stack: ItemStack?, slotId: Int): INetworkItem? {
        return WirelessFluidGridNetworkItem(handler!!, player!!, stack!!, slotId)
    }

    companion object {
        @kotlin.jvm.JvmStatic
        fun getSortingType(stack: ItemStack): Int {
            return if (stack.hasTag() && stack.tag!!.contains(GridNetworkNode.NBT_SORTING_TYPE)) stack.tag!!.getInt(GridNetworkNode.NBT_SORTING_TYPE) else IGrid.SORTING_TYPE_QUANTITY
        }

        @kotlin.jvm.JvmStatic
        fun getSortingDirection(stack: ItemStack): Int {
            return if (stack.hasTag() && stack.tag!!.contains(GridNetworkNode.NBT_SORTING_DIRECTION)) stack.tag!!.getInt(GridNetworkNode.NBT_SORTING_DIRECTION) else IGrid.SORTING_DIRECTION_DESCENDING
        }

        @kotlin.jvm.JvmStatic
        fun getSearchBoxMode(stack: ItemStack): Int {
            return if (stack.hasTag() && stack.tag!!.contains(GridNetworkNode.NBT_SEARCH_BOX_MODE)) stack.tag!!.getInt(GridNetworkNode.NBT_SEARCH_BOX_MODE) else IGrid.SEARCH_BOX_MODE_NORMAL
        }

        @kotlin.jvm.JvmStatic
        fun getTabSelected(stack: ItemStack): Int {
            return if (stack.hasTag() && stack.tag!!.contains(GridNetworkNode.NBT_TAB_SELECTED)) stack.tag!!.getInt(GridNetworkNode.NBT_TAB_SELECTED) else -1
        }

        @kotlin.jvm.JvmStatic
        fun getTabPage(stack: ItemStack): Int {
            return if (stack.hasTag() && stack.tag!!.contains(GridNetworkNode.NBT_TAB_PAGE)) stack.tag!!.getInt(GridNetworkNode.NBT_TAB_PAGE) else 0
        }

        @kotlin.jvm.JvmStatic
        fun getSize(stack: ItemStack): Int {
            return if (stack.hasTag() && stack.tag!!.contains(GridNetworkNode.NBT_SIZE)) stack.tag!!.getInt(GridNetworkNode.NBT_SIZE) else IGrid.SIZE_STRETCH
        }
    }

    init {
        this.setRegistryName(RS.ID, (if (type == Type.CREATIVE) "creative_" else "") + "wireless_fluid_grid")
    }
}